src/youtube-player/youtube-player.spec.ts

```diff
@@ -0,0 +1,121 @@
+import {async, ComponentFixture, TestBed} from '@angular/core/testing';
+import {Component, ViewChild} from '@angular/core';
+import {By} from '@angular/platform-browser';
+import {YouTubePlayerModule} from './index';
+import {YouTubePlayer} from './youtube-player';
+
+const VIDEO_ID = 'a12345';
+
+const playerState = {
+  UNSTARTED: -1,
+  CUED: 5,
+  OTHER: 9,
+};
+
+describe('YoutubePlayer', () => {
+  let playerCtor: jasmine.Spy;
+  let player: jasmine.SpyObj<YT.Player>;
+  let onPlayerReady: () => void;
+  let fixture: ComponentFixture<TestApp>;
+  let testComponent: TestApp;
+
+  beforeEach(async(() => {
+    playerCtor = jasmine.createSpy('Player Constructor');
```

Comment:
Would it make sense to just make a standalone `FakeYouTubePlayer`?

Reply:
Could you provide some reasoning? There's quite a few methods that need to be spied on/mocked.